zebra-state/src/service/non_finalized_state/queued_blocks.rs

```diff
@@ -48,32 +52,40 @@ impl QueuedBlocks {
             .insert(new_hash);
         assert!(inserted, "hashes must be unique");
 
-        tracing::trace!(num_blocks = %self.blocks.len(), %parent_hash, ?new_height,  "Finished queueing a new block");
+        tracing::trace!(%parent_hash, queued = %self.blocks.len(), "queued block");
         self.update_metrics();
     }
 
     /// Dequeue and return all blocks that were waiting for the arrival of
     /// `parent`.
-    #[instrument(skip(self))]
-    pub fn dequeue_children(&mut self, parent: block::Hash) -> Vec<QueuedBlock> {
+    #[instrument(skip(self), fields(%parent_hash))]
+    pub fn dequeue_children(&mut self, parent_hash: block::Hash) -> Vec<QueuedBlock> {
         let queued_children = self
             .by_parent
-            .remove(&parent)
+            .remove(&parent_hash)
             .unwrap_or_default()
             .into_iter()
             .map(|hash| {
-                self.blocks
+                let queued = self
+                    .blocks
                     .remove(&hash)
-                    .expect("block is present if its hash is in by_parent")
+                    .expect("block is present if its hash is in by_parent");
+                queued
```

Comment:
Stable clippy says...
```suggestion
                    .expect("block is present if its hash is in by_parent")
```